// truedata-nodejs/utils.js
const L2arr = ['SymbolId', 'Time', 'SequenceNo'];
const L2common = ['Bid', 'BidQty', 'BidNo', 'Ask', 'AskQty', 'AskNo'];
for (let i = 1; i < 6; i++) {
	L2common.forEach((L) => L2arr.push(`${L}${i}`));
}
L2arr.push('TotalBidQty', 'TotalAskQty');

module.exports.L2Keys = L2arr;
module.exports.expires_in = 3600 - 20; // - 20 is to make sure user gets new token 20 seconds before current token expires
module.exports.reconnectionTime = 2000;
module.exports.bidDeactivatedMessage = 'Deactivated';

module.exports.root = 'https://history.truedata.in';
module.exports.login = 'https://auth.truedata.in/token';

module.exports.routes = {
	getTickData: '/getticks',
	getBarData: '/getbars',
	getLastNBars: '/getlastnbars',
	getLastNTicks: '/getlastnticks',
	getBhavCopy: '/getbhavcopy',
	getBhavCopyStatus: '/getbhavcopystatus',
	getTopGainers: '/gettopngainers',
	getTopLosers: '/gettopnlosers',
	getTopVolumeGainers: '/gettopnvolumegainers',
	getCorpAction: '/getcorpaction',
	getSymbolNameChange: '/getsymbolchangehistory',
};

module.exports.formatTime = (year, month, date, hour, min) => {
	return `${year < 100 ? year : year - 2000}${zeroChecker(month)}${zeroChecker(date)}T${zeroChecker(
		hour
	)}:${zeroChecker(min)}:00`;
};

function zeroChecker(number) {
	return number < 10 ? `0${number}` : number;
}

async function wait(timeout) {
	new Promise((res, rej) => setTimeout(res(), timeout));
}

module.exports.wait = wait;
module.exports.zeroChecker = zeroChecker;
